src/metabase/api/collection.clj

```diff
@@ -75,25 +75,31 @@
   the root, if `collection-id` is `nil`) and its immediate children, to avoid reading the entire collection tree when it
   is not necessary.
 
-  For archived, we can either pass in include both (when archived is nil), include only archived is true, or archived is false."
-  [archived exclude-other-user-collections namespace shallow collection-id]
+  For archived, we can either include everthing (when archived is `nil`), only archived (when `archived` is true),
+  or only non-archived (when `archived` is false).
+
+  To select only personal collections, pass in `personal-only` as `true`.
+  This will select only collections where `personal_owner_id` is not `nil`."
+  [{:keys [archived exclude-other-user-collections namespace shallow collection-id personal-only permissions-set]}]
   (cond->>
    (t2/select :model/Collection
               {:where [:and
                        (when (some? archived)
                          [:= :archived archived])
                        (when shallow
                          (location-from-collection-id-clause collection-id))
+                       (when personal-only
+                         [:!= :personal_owner_id nil])
                        (when exclude-other-user-collections
                          [:or [:= :personal_owner_id nil] [:= :personal_owner_id api/*current-user-id*]])
                        (perms/audit-namespace-clause :namespace namespace)
                        (collection/visible-collection-ids->honeysql-filter-clause
                         :id
-                        (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))]
+                        (collection/permissions-set->visible-collection-ids permissions-set))]
```

Comment:
I'm worried that someone might not realize that you're expected to pass in `permissions-set` here, and forget to do it, and have something bad happen... I think having a default behavior keeps someone from doing something dangerous and also makes this a little nicer to use from the REPL and some tests. Then you don't need to make some of the changes to existing usages

```suggestion
                        (collection/permissions-set->visible-collection-ids (or permissions-set @api/*current-user-permissions-set*)))]
```